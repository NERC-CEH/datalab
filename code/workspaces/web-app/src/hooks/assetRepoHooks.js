import useShallowSelector from './useShallowSelector';
import assetRepoSelectors from '../selectors/assetRepoSelectors';
import sortByName from '../components/common/sortByName';
import { PUBLIC } from '../components/assetRepo/assetVisibilities';

const useAssetRepoUnsorted = () => useShallowSelector(assetRepoSelectors.assetRepo);

export const useAssetRepo = () => {
  const assetRepo = useAssetRepoUnsorted();
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(assetRepo.value.assets),
    },
  };
};

const assetVisible = (projectKey, asset) => asset.fileLocation
  && (asset.visible === PUBLIC
  || (asset.projects && asset.projects.map(project => project.key).includes(projectKey)));

export const useVisibleAssets = (projectKey) => {
  const assetRepo = useAssetRepoUnsorted();
  const visibleAssets = assetRepo.value.assets
    ? assetRepo.value.assets.filter(asset => assetVisible(projectKey, asset))
    : [];
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(visibleAssets),
    },
  };
};
