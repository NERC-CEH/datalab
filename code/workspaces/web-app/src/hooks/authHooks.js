import useShallowSelector from './useShallowSelector';
import authSelectors from '../selectors/authSelectors';

const { currentUserId, currentUserPermissions, currentUserTokens } = authSelectors;

export const useCurrentUserId = () => useShallowSelector(currentUserId);
export const useCurrentUserPermissions = () => useShallowSelector(currentUserPermissions);
export const useCurrentUserTokens = () => useShallowSelector(currentUserTokens);
