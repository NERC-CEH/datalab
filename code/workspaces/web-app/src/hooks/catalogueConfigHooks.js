import useShallowSelector from './useShallowSelector';
import catalogueConfigSelectors from '../selectors/catalogueConfigSelectors';

export const useCatalogueConfig = () => useShallowSelector(catalogueConfigSelectors.catalogueConfig);
export const useCatalogueAvailable = () => useShallowSelector(catalogueConfigSelectors.catalogueAvailable);
