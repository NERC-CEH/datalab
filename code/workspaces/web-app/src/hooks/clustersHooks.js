import useShallowSelector from './useShallowSelector';
import clustersSelectors from '../selectors/clustersSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useClusters = () => useShallowSelector(clustersSelectors.clusters);
export const useClustersByType = (type) => {
  const clusters = useClusters();

  if (!type) {
    return clusters;
  }

  return {
    ...clusters,
    value: clusters.value.filter(cluster => cluster.type === type),
  };
};
