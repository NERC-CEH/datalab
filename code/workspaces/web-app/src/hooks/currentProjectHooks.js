import useShallowSelector from './useShallowSelector';
import currentProjectSelectors from '../selectors/currentProjectSelectors';

const { currentProject, currentProjectKey } = currentProjectSelectors;

export const useCurrentProject = () => useShallowSelector(currentProject);
export const useCurrentProjectKey = () => useShallowSelector(currentProjectKey);
