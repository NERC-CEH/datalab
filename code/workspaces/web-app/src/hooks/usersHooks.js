import useShallowSelector from './useShallowSelector';
import usersSelectors from '../selectors/usersSelectors';
import sortByName from '../components/common/sortByName';

export const useUsers = () => useShallowSelector(usersSelectors.users);

export const useUsersSortedByName = () => {
  const users = useUsers();
  return {
    ...users,
    value: sortByName(users.value),
  };
};
