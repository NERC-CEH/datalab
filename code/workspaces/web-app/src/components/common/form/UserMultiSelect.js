import React, { useState } from 'react';
import { renderMultiSelectAutocompleteField } from './controls';
import { useUsers } from '../../../hooks/usersHooks';
import sortByName from '../sortByName';

function UserMultiSelect({ input, meta = null, ...custom }) {
  const users = useUsers();
  const sortedUsers = sortByName(users.value);
  const [currentValue, setCurrentValue] = useState(input.value || []); // use to give current value to onBlur

  return (
    <>
      {
      renderMultiSelectAutocompleteField(
        {
          input,
          meta,
          currentValue,
          setCurrentValue,
          options: sortedUsers,
          label: 'Users',
          placeholder: "Type user's email address",
          getOptionLabel: val => val.name,
          getOptionSelected: (option, val) => option.userId === val.userId,
          loading: users.fetching,
          selectedTip: 'User selected',
          ...custom,
        },
      )
    }
    </>
  );
}

export default UserMultiSelect;
