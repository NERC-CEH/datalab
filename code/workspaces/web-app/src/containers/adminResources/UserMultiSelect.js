import React from 'react';
import { renderMultiselectAutocompleteField } from '../../components/common/form/controls';
import { useUsers } from '../../hooks/usersHooks';
import sortByName from './sortByName';

function UserMultiSelect(props) {
  const { selectedUsers, setSelectedUsers } = props;
  const users = useUsers();

  const sortedUsers = sortByName(users.value);

  return (
    <>
      {
        renderMultiselectAutocompleteField(
          {
            input: {
              onChange: (val) => { setSelectedUsers(val); },
              value: selectedUsers,
            },
            options: sortedUsers,
            label: 'Users',
            placeholder: 'Filter by user',
            getOptionLabel: val => val.name,
            getOptionSelected: val => selectedUsers.includes(val),
            loading: users.fetching,
            selectedTip: 'User selected',
          },
        )
      }
    </>
  );
}

export default UserMultiSelect;
