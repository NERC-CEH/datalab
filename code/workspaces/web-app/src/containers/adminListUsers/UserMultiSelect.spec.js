import React from 'react';
import { useDispatch } from 'react-redux';
import { shallow } from 'enzyme';
import { useUsers } from '../../hooks/usersHooks';
import UserMultiSelect from './UserMultiSelect';

jest.mock('react-redux');
jest.mock('../../hooks/usersHooks');

const user1 = { userId: 'user-1', name: 'Test Name 1' };
const user2 = { userId: 'user-2', name: 'Test Name 2' };

useDispatch.mockReturnValue(jest.fn().mockName('dispatch'));
useUsers.mockReturnValue({ fetching: false, value: [user1, user2] });

describe('UserMultiSelect', () => {
  const shallowRender = () => {
    const props = {
      selectedUsers: [user1, user2],
      setSelectedUsers: jest.fn().mockName('set-selected-user'),
    };

    return shallow(<UserMultiSelect {...props} />);
  };

  it('renders to match snapshot passing correct props to children', () => {
    expect(shallowRender()).toMatchSnapshot();
  });
});
