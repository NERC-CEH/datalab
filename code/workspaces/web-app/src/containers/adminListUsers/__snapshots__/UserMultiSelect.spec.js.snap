// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMultiSelect renders to match snapshot passing correct props to children 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Autocomplete))
    autoHighlight={true}
    getOptionLabel={[Function]}
    getOptionSelected={[Function]}
    loading={false}
    multiple={true}
    onChange={[Function]}
    options={
      Array [
        Object {
          "name": "Test Name 1",
          "userId": "user-1",
        },
        Object {
          "name": "Test Name 2",
          "userId": "user-2",
        },
      ]
    }
    renderInput={[Function]}
    renderOption={[Function]}
    style={
      Object {
        "marginBottom": 10,
        "minWidth": 250,
      }
    }
    value={
      Array [
        Object {
          "name": "Test Name 1",
          "userId": "user-1",
        },
        Object {
          "name": "Test Name 2",
          "userId": "user-2",
        },
      ]
    }
  />
</Fragment>
`;
